pkg/controllers/job/job_controller_actions.go

```diff
@@ -199,6 +176,10 @@ func (cc *Controller) syncJob(jobInfo *apis.JobInfo, updateStatus state.UpdateSt
 		return nil
 	}
 
+	if err := cc.createJob(job); err != nil {
```

Comment:
We also need to skip creating pods here if pod group is not ready, creating and deny in admission service would be a longer process.

Reply:
syncJob should be pure, need less if